cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -843,6 +859,74 @@ impl ManagedEnvironment {
         Ok(result)
     }
 
+    /// Create a new generation from local changes,
+    /// and updates the generation lock.
+    ///
+    /// If no changes exist, returns without further action.
+    ///
+    /// TODO: this should do a `build` before committing to a generation
+    pub fn sync_local(&self, flox: &Flox) -> Result<(), ManagedEnvironmentError> {
+        let mut local = self.local_checkout(flox)?;
+
+        if Self::validate_checkout(&local, &self.get_current_generation(flox)?)? {
+            debug!("local checkout and remote checkout equal, nothing to apply");
+            return Ok(());
+        }
+
+        let mut generations = self
+            .generations()
+            .writable(flox.temp_dir.clone())
+            .map_err(ManagedEnvironmentError::CreateFloxmetaDir)?;
+
+        generations
+            .add_generation(
+                &mut local,
+                "Synchronized manual changes to generation".to_string(),
+            )
+            .map_err(ManagedEnvironmentError::CommitGeneration)?;
+
+        self.lock_pointer()?;
+        Ok(())
+    }
+
+    /// Create or reset a local checkout of the latet generation.
+    ///
+    /// Copies the `env/` directory from the latest generation to the `.flox/` directory
```

Comment:
Question: so if there are files already in `.flox` the contents of those files will just be lost forever, correct?